### ISSUE TITLE
Respond to the HashiCorp license change

### ISSUE THREAD

**Issue by JordanJordanov** [gardener/gardener#8377]
**How to categorize this issue?**

/area delivery
/kind enhancement

**What would you like to be added**:
HashiCorp announced a [license change](https://www.hashicorp.com/blog/hashicorp-adopts-business-source-license) which will not be permitted within CNCF (nor SAP). For that reason [CNCF community is collecting HashiCorp components usage](https://github.com/cncf/foundation/issues/617) to prepare for the proper response. 
Just like a lot of other Kubernetes components, in Gardener [we are also using several (more or less standard) Go libraries from Hashicorp](https://github.com/search?q=org%3Agardener%20HashiCorp%20&type=code) for which I am opening this issue.

At this moment it's too early to think about concrete migration steps - we shall wait for the response from the community first (e.g. certain projects might be forked).

**Why is this needed**: 
Compliance: use libraries with permitted license

According to HashiCorp ([discussion forum](https://discuss.hashicorp.com/t/hashicorp-projects-changing-license-to-business-source-license-v1-1/57106), [FAQ](https://www.hashicorp.com/license-faq#products-covered-by-bsl)) the following products/projects and versions are affected:

- [Terraform](https://github.com/hashicorp/terraform) >1.5.5
- [Vagrant](https://github.com/hashicorp/vagrant) >2.3.7
- [Packer](https://github.com/hashicorp/packer) >1.9.2
- [Waypoint](https://github.com/hashicorp/waypoint) >0.11.4
- [Boundary](https://github.com/hashicorp/boundary) >0.13.1
- [Consul](https://github.com/hashicorp/consul) >1.16.1
- [Nomad](https://github.com/hashicorp/nomad) >1.6.1
- [Vault](https://github.com/hashicorp/vault) >1.14.1
- [Vault Secrets Operator](https://github.com/hashicorp/vault-secrets-operator) > 0.1.0
- [Vault Provider for Secrets Store CSI Driver](https://github.com/hashicorp/vault-csi-provider) > 1.4.0
- [Go library for encrypting values through various KMS providers](https://github.com/hashicorp/go-kms-wrapping) license change was announced, but has not yet happened (Aug 16, 2023)

**Comment by rfranzke** [comment]
We only use `terraform`, but the versions are controlled in https://github.com/gardener/terraformer (e.g., [here](/blob/master/build/all/TF_VERSION)), and generally the colleagues (@kon-angelo @MartinWeindel) are already in the process of migrating away from `terraform`.

In `gardener/gardener`, the only direct dependency to a HashiCorp library is https://github.com/gardener/gardener/blob/27bb801d262127e62bbf3a67bf556f0f216ef1f6/go.mod#L21 I don't think this affected by this license change.

Other than those, nothing else from your list above is used in `gardener/gardener`.

TL;DR: Shall we move this issue to https://github.com/gardener/terraformer (or maybe simply close it entirely since I don't see what concrete action items should be taken now)?
